src/MoviePage/MoviePage.js

```diff
@@ -1,41 +1,41 @@
-import React from "react";
-import { getIndividualMovie } from '../api';
+import React, {Component} from "react";
+import {getIndividualMovie} from "../api";
 
-function MoviePage(props) {
-  let id = props.location.pathname.split("/")[2];
-  
-  if (props.match.isExact) {
-    console.log(props);
-    const {
-      average_rating,
-      backdrop_path,
-      budget,
-      genres,
-      id,
-      overview,
-      poster_path,
-      release_date,
-      revenue,
-      runtime,
-      tagline,
-      title,
-    } = props.location.state.movie;
+class MoviePage extends Component {
+  constructor(props) {
+    super();
```

Comment:
While I'm not 100% sold on the class component here, it just makes more sense to do it like this for now

Reply:
Yeah I think there might be a blindspot that we can do more research on since we're also having difficulty testing functions that use `history.push` 

I'm wondering if we can make use of `match` and dynamic  paths more to resolve both testing + re-converting `MoviePage` to a functional component